ql/src/java/org/apache/hadoop/hive/ql/exec/FileSinkOperator.java

```diff
@@ -251,7 +271,7 @@ public void closeWriters(boolean abort) throws HiveException {
       }
     }
 
-    private void commit(FileSystem fs, List<Path> commitPaths) throws HiveException {
+    private void commit(FileSystem fs, List<Path> commitPaths, List<Path> deleteDeltas) throws HiveException {
```

Comment:
Think through 1 more time. I can accept that this is the best solution, but this is UGLY 😄

Reply:
I know, but I don't really know a better solution, only if we change the internal structures in FileSinkOperator. Like using Lists instead of arrays. But this could have unexpected side effects. I am open to try it but I would do it under a separate Jira. I create one about investigating this refactoring.